src/components/charts/useChartGroup.ts

```diff
@@ -225,25 +236,35 @@ export default function useChartGroup(chartGroupId = null): ChartGroupExports {
             showError(err)
         }
     }
-    
+
     async function validateData() {
         try {
             const nameRegexp = new RegExp(`^[a-z]+[a-z0-9\-\?]*[a-z0-9]+$`)
+            
+            const allNames = state.charts.map((chart) => chart.name.value)
+            let isNotValid = allNames.filter((name, index) => {
+                if (allNames.indexOf(name) != index) {
+                    return index
+                }
+                return false
+            })
             let validated = true
             let tempCharts = state.charts.map((chart) => {
                 if (!chart.isEnabled) {
                     // dont consider disabled charts
                     return chart
                 }
-                if (!nameRegexp.test(chart.name.value) || !chart?.environment?.id) {
+                if (!nameRegexp.test(chart.name.value) || !chart?.environment?.id || isNotValid?.length > 0) {
                     validated = false
                 }
                 return {
                     ...chart,
                     name: {
                         value: chart.name.value,
                         error: nameRegexp.test(chart.name.value)
-                            ? ''
+                            ? isNotValid.indexOf(chart.name.value) === -1
```

Comment:
Avoid ternary nesting instaed extract out in a separate function for more rediability